umm-spec-lib/src/cmr/umm_spec/umm_to_xml_mappings/iso19115_2/spatial.clj

```diff
@@ -86,7 +88,42 @@
         shapes (spatial-lib-shapes (-> spatial :HorizontalSpatialDomain :Geometry))]
     (->> shapes
          (map (partial umm-s/set-coordinate-system coordinate-system))
-         (map d/calculate-derived)
-         ;; ISO MENDS interleaves MBRs and actual spatial areas
-         (mapcat (juxt r/mbr identity))
          (map gmd/encode))))
+
+(defn generate-zone-identifier
+  "Returns a geographic element for the zone identifier"
+  [c]
+  (when-let [zone-identifier (-> c :SpatialExtent :HorizontalSpatialDomain :ZoneIdentifier)]
```

Comment:
I'm not sure if you'll notice a performance difference either way, but I think `get-in` is preferred here. I do like the threading macro style-wise, though.
`(get-in c [:SpatialExtent :HorizontalSpatialDomain :ZoneIdentifier])`